First set of UI tweaks

Iemand005 opened the pull request:
# Fixes some of the UI design errors in ALVR Dashboard.

## Before

<img width="902" height="632" alt="afbeelding" src="https://github.com/user-attachments/assets/f72d40d3-afae-49bb-bab8-306cee2ea7e3" />

## After

<img width="942" height="632" alt="afbeelding" src="https://github.com/user-attachments/assets/6b9f219a-46e1-41f2-a521-cc9fce459d31" />

Changes:
- Square reset buttons
- Centered toggle buttons
- Centered scroll bar
- Removed odd margin above scroll bar's value display
- Adjusted panel corner radius to make the margin in the rounding even with the buttons
- Scrollbar value display drag speed set to linear

Let me know what else needs to be improved. Thank you very much!

zmerp commented on alvr/gui_common/src/basic_components/switch.rs:
```diff
@@ -1,7 +1,7 @@
 
 pub fn switch(ui: &mut Ui, on: &mut bool) -> Response {
-    let desired_size = ui.spacing().interact_size.y * egui::vec2(2.0, 1.0);
+    let desired_size = 20.0 * egui::vec2(2.0, 1.0);
```
The "20" magic number appears here again, you should put the constant in theme.rs and explain exactly what it means

zmerp commented on alvr/gui_common/src/theme.rs:
```diff
@@ -59,6 +62,7 @@ pub fn set_theme(ctx: &Context) {
     let mut visuals = Visuals::dark();
 
     let corner_radius = CornerRadius::same(10);
+    let frame_corner_radius = CornerRadius::same(15);
```
This value could probably be inlined later at line 86

Iemand005 commented on alvr/dashboard/src/dashboard/components/settings_controls/number.rs:
```diff
@@ -78,45 +78,53 @@ impl Control {
-                        textbox = textbox.suffix(suffix);
-                    }
+                ui.scope(|ui| {
+                    ui.style_mut().spacing.interact_size.y = 20.0;
```
Yup I was gonna ask you what method I should use to declare the constants in theme.rs. I could put static variables that can be imported or a struct or something.

zmerp replied:
Just put the constant somewhere in theme.rs, no scope needed unless you want to make a family of constants (eg colors). Also use `const`, not `static`

Iemand005 replied:
Alright thank you, I updated the code, I think the variable names are enough explain what the magic value does. Let me know what else needs to be changed.

Iemand005 commented on alvr/dashboard/src/dashboard/components/settings_controls/reset.rs:
```diff
@@ -1,14 +1,18 @@
 
-        ui.add_enabled(enabled, Button::new("⟲"))
-            .on_hover_text(format!("Reset to {default_str}"))
+        ui.scope(|ui| {
```
Ah no that was left over while testing stuff, it's not necessary anymore as the interact_size from theme.rs works now. I forgot to remove this. Thanks for reminding me!

Iemand005 commented on alvr/dashboard/src/dashboard/components/settings_controls/reset.rs:
```diff
@@ -1,14 +1,18 @@
-            .on_hover_text(format!("Reset to {default_str}"))
+        ui.scope(|ui| {
+            let height = ui.spacing().interact_size.y;
+            ui.add_enabled(enabled, Button::new("⟲").min_size(vec2(height, height)))
```
Will do that from now on!

zmerp commented on alvr/gui_common/src/theme.rs:
```diff
@@ -49,7 +52,10 @@ pub fn set_theme(ctx: &Context) {
 
     let mut style = (*ctx.style()).clone();
-    style.spacing.slider_width = 200_f32; // slider width can only be set globally
+    style.spacing.slider_width = 200_f32; // slider width can only be set globally, but you can use ui.scope to override it locally
```
remove the comment about ui.scope

Iemand005 commented on alvr/gui_common/src/theme.rs:
```diff
@@ -79,8 +80,11 @@ pub fn set_theme(ctx: &Context) {
-    visuals.widgets.noninteractive.bg_fill = BG;
     visuals.faint_bg_color = DARKER_BG;
+
+    let frame_corner_radius = CornerRadius::same(15);
```
The radius of the frame should be the button radius + the margin around the buttons inside it. I'll check where that's defined, I didn't see it in theme.rs it might be hard coded, if it is I'll turn that into a const variable in theme.rs too and express it with maths.

Iemand005 commented on alvr/dashboard/src/dashboard/components/settings.rs:
```diff
@@ -123,13 +123,11 @@ impl SettingsTab {
-            let visuals = ui.style().visuals.clone();
-            let margin = 12.0;
-            let corner_radius = visuals.widgets.open.corner_radius.average() + margin;
+            let corner_radius = ui.style().visuals.widgets.open.corner_radius.average() + SETTINGS_TAB_FRAME_MARGIN;
```
It's currently only used in the settings page, I'll move it to there and if someone ever adds tabs to another page it can be moved to `theme.rs`

Iemand005 commented on alvr/dashboard/src/dashboard/components/devices.rs:
```diff
@@ -64,10 +64,7 @@ impl DevicesTab {
                         ui.horizontal(|ui| {
                             ui.add_space(10.0);
```
The margin seems to be hard coded here and probably in the other places too. I think it's better for this to be placed in `theme.rs`, then we can use maths to calculate what the frame corner radius should be.

Iemand005 commented on alvr/dashboard/src/dashboard/components/settings_controls/number.rs:
```diff
@@ -78,45 +78,53 @@ impl Control {
-                    // todo: investigate why the slider does not get centered vertically
-                    ui.with_layout(Layout::left_to_right(Align::Center), |ui| ui.add(slider))
-                        .inner
+            let mut is_editing = false;
+            let mut finished_editing = false;
```
The error was: cannot borrow `*self.editing_value_f64` as mutable more than once at a time
second mutable borrow occurs here

zmerp replied:
If you have problems with this let me know if i should work on this myself

Iemand005 commented:
I extracted the hard coded values and updated the frames to respond to the padding setting. The corner radius of frames is also automatically calculated to match whatever the other values are set to.


## Padding 10:

<img width="902" height="632" alt="afbeelding" src="https://github.com/user-attachments/assets/ec97bada-bcf8-492d-af37-ef70ae40b4e5" />
<img width="902" height="632" alt="afbeelding" src="https://github.com/user-attachments/assets/988f8479-f058-4377-9676-c613df546cff" />
<img width="902" height="632" alt="afbeelding" src="https://github.com/user-attachments/assets/78bfc8e0-8667-493a-9d3c-04a38628973a" />


## Padding 30:

<img width="1054" height="673" alt="afbeelding" src="https://github.com/user-attachments/assets/89f09a3a-b378-4a37-a8ba-582deb4952cd" />
<img width="1054" height="673" alt="afbeelding" src="https://github.com/user-attachments/assets/4ef46c49-269d-4409-8aa3-f75b44c4dac6" />
<img width="1054" height="673" alt="afbeelding" src="https://github.com/user-attachments/assets/6106ff99-78ac-4be8-af99-acad4c7defea" />


Settings frame is updated to have the same padding and rounding as the other frames:

<img width="560" height="149" alt="afbeelding" src="https://github.com/user-attachments/assets/85db433f-31f8-4e35-b057-8ea55d1e820c" />

zmerp commented:
Keep padding to 10

zmerp commented:
@Iemand005 you need to fix the code in number.rs or let me know if I should take care of it

The-personified-devil commented on alvr/session/src/settings.rs:
```diff
@@ -1414,6 +1421,7 @@ This could happen on TCP. A IDR frame is requested in this case."#
     #[schema(gui(slider(min = 5, max = 1000, step = 5)), suffix = "ms")]
     pub minimum_idr_interval_ms: u64,
 
+    #[schema(strings(display_name = "DSCP"))]
```
this could be a lot more descriptive, like `DSCP (packet prioritization hints)`, which still means nothing to the average person, but it's an improvement

zmerp replied:
Made it a bit shorter, to avoid shifting the labels columns too far to the right

zmerp commented on alvr/dashboard/src/dashboard/components/devices.rs:
```diff
@@ -58,16 +58,14 @@ impl DevicesTab {
-                                    "ALVR requires running SteamVR! {}",
-                                    "Devices will not be discovered or connected"
-                                ))
+                                RichText::new("ALVR requires running SteamVR! Devices will not be discovered or connected.")
```
there is a bunch of instances where this can be done. but apart from this instance let's not hunt for the other strings in this PR
